lib/client/doc.js

```diff
@@ -342,13 +345,15 @@ Doc.prototype._handleOp = function(err, message) {
     return;
   }
 
-  if (this.inflightOp) {
-    var transformErr = transformX(this.inflightOp, message);
-    if (transformErr) return this._hardRollback(transformErr);
+  var opsToTransform = [];
+  if (this.inflightOp) opsToTransform.push(this.inflightOp);
+  opsToTransform = opsToTransform.concat(this.pendingOps);
+  for (var id in this._heldOps) {
```

Comment:
This is where my understanding of OT starts to get pushed a bit — does the transformation ordering matter here? I'd have normally assumed no, but I think `transformX` mutates the server op, so order **does** matter (I confirmed this trivially by moving the inflight op to after pending, and a couple of tests broke). I just don't know how (because held ops are temporally separate from pending ops). It could be that it's only important for `create` and `del`?